library-model/library-model-generator/src/test/java/com/uber/nullaway/libmodel/LibraryModelGeneratorTest.java

```diff
@@ -86,6 +98,190 @@ public void nullableUpperBound() throws IOException {
         };
     ImmutableMap<String, Set<Integer>> expectedNullableUpperBounds =
         ImmutableMap.of("NullableUpperBound", ImmutableSet.of(0));
-    runTest("NullableUpperBound.java", lines, ImmutableMap.of(), expectedNullableUpperBounds);
+    runTest(
+        "NullableUpperBound.java",
+        lines,
+        ImmutableMap.of(),
+        expectedNullableUpperBounds,
+        ImmutableSet.of("NullableUpperBound"));
+  }
+
+  @Test
+  public void nullMarkedClasses() throws IOException {
+    String[] lines =
+        new String[] {
+          "import org.jspecify.annotations.NullMarked;",
+          "@NullMarked",
+          "public class NullMarked {",
+          "  public static class Nested {}",
+          "}",
+        };
+    ImmutableSet<String> expectedNullMarkedClasses =
+        ImmutableSet.of("NullMarked", "NullMarked.Nested");
+    runTest(
+        "NullMarked.java", lines, ImmutableMap.of(), ImmutableMap.of(), expectedNullMarkedClasses);
+  }
+
+  @Test
+  public void noNullMarkedClasses() throws IOException {
+    String[] lines =
+        new String[] {
+          "import org.jspecify.annotations.NullMarked;",
+          "public class NotNullMarked {",
+          "  public static class Nested {}",
+          "}",
+        };
+    ImmutableSet<String> expectedNullMarkedClasses = ImmutableSet.of();
+    runTest(
+        "NotNullMarked.java",
+        lines,
+        ImmutableMap.of(),
+        ImmutableMap.of(),
+        expectedNullMarkedClasses);
+  }
+
+  @Test
+  public void nullableParameters() throws IOException {
+    String[] lines =
+        new String[] {
+          "import org.jspecify.annotations.NullMarked;",
+          "import org.jspecify.annotations.Nullable;",
+          "@NullMarked",
+          "public class NullableParameters{",
+          " public static Object getNewObjectIfNull(@Nullable Object object) {",
+          "   if (object == null) {",
+          "     return new Object();",
+          "   } else {",
+          "     return object;",
+          "   }",
+          " }",
+          "}"
+        };
+    ImmutableMap<String, MethodAnnotationsRecord> expectedMethodRecords =
+        ImmutableMap.of(
+            "NullableParameters:java.lang.Object getNewObjectIfNull(java.lang.Object)",
+            MethodAnnotationsRecord.create(
+                ImmutableSet.of(), ImmutableMap.of(0, ImmutableSet.of("Nullable"))));
+    runTest(
+        "NullableParameters.java",
+        lines,
+        expectedMethodRecords,
+        ImmutableMap.of(),
+        ImmutableSet.of("NullableParameters"));
+  }
+
+  @Test
+  public void nullableParametersInNullUnmarkedClass() throws IOException {
```

Comment:
This is incorrectly testing a `@NullMarked` class

Reply:
Oh, my bad. I have an additional question regarding this test, we would not be creating an astubx if the class in not @NullMarked correct? so it will fail on `astubx file was not created` as per our current logic right?

Reply:
Won't fail see the test `noNullMarkedClasses` which passes